django_filters/fields.py

```diff
@@ -257,22 +265,32 @@ def __init__(self, *args, **kwargs):
 
         super().__init__(*args, **kwargs)
 
-    def _get_choices(self):
-        return super()._get_choices()
+    if not DJANGO_50:
+
+        def _get_choices(self):
+            return super()._get_choices()
 
-    def _set_choices(self, value):
-        super()._set_choices(value)
-        value = self.iterator(self, self._choices)
+        def _set_choices(self, value):
+            super()._set_choices(value)
+            value = self.iterator(self, self._choices)
 
-        self._choices = self.widget.choices = value
+            self._choices = self.widget.choices = value
 
-    choices = property(_get_choices, _set_choices)
+        choices = property(_get_choices, _set_choices)
 
 
 # Unlike their Model* counterparts, forms.ChoiceField and forms.MultipleChoiceField do not set empty_label
 class ChoiceField(ChoiceIteratorMixin, forms.ChoiceField):
     iterator = ChoiceIterator
 
+    if DJANGO_50:
+
+        @forms.ChoiceField.choices.setter
+        def choices(self, value):
+            self._choices = self.widget.choices = self.iterator(
+                self, normalize_choices(value)
```

Comment:
There are a few interesting things here:

1. Ideally we'd not rehash the Django internals here, i.e. `self._choices = self.widget.choices = ...`
2. For equivalent behaviour as in Django 5.0 it ought to be `... = normalize_choices(self.iterator(...))`
3. To make the normalization lazy, `ChoiceIterator` should extend `django.utils.choices.ChoiceIterator`
    - _Maybe I should have called that `django.utils.choices.BaseChoiceIterator`?_ 🤔 
4. We should push the `normalize_choices()` call into `ChoiceIterator.__iter__()`
5. It's a shame that we have to implement this twice rather than once in the mixin

I think all of this can be solved. Will open an alternate PR soon...